src/main/java/au/org/ala/biocache/util/QueryFormatUtils.java

```diff
@@ -149,15 +150,15 @@ public Map[] formatSearchQuery(SpatialSearchRequestDTO searchParams, boolean for
             searchParams.setFormattedFq(null);
 
             // Apply filter tagging and excluding filters if flag is set
-            if (searchParams.getIncludeUnfilteredFacetValues()) {
+            if (isIncludeUnfilteredFacetValues) {
                 applyFilterTagging(searchParams);
             }
 
             //format fqs for facets that need ranges substituted
             if (searchParams.getFq() != null) {
                 for (int i = 0; i < searchParams.getFq().length; i++) {
                     String fq = searchParams.getFq()[i];
-                    String fqOriginal = originalFqs[i]; // not altered by `applyFilterTagging()`
+                    String fqOriginal = (originalFqs.length > i) ? originalFqs[i] : ""; // not altered by `applyFilterTagging()`
```

Comment:
When the the originalFqs[i] is absent, instead of `empty string`, use `fq`.

Reply:
Agree - I'm adding some unit tests that catch the `invalid array index` error, so will include those with next commit/fix.